File: performance-test/generators.py
from common import fake
import uuid


def random_user():
    return {
        'id': str(uuid.uuid4()),
        'name': fake.user_name(),
    }


def random_book():
    return {
        'id': str(uuid.uuid4()),
        'title': " ".join(fake.words()),
        'author': fake.simple_profile()['name']
    }


def random_offer(bookId, sellerId):
    return {
        'id': str(uuid.uuid4()),
        'bookId': bookId,
        'sellerId': sellerId,
        'price': fake.pyfloat(min_value=1.0, max_value=99.0),
        'stock': fake.pyint(min_value=1)
    }

File: k6/generators.js
import {
    randomIntBetween,
    randomString,
    uuidv4,
} from 'https://jslib.k6.io/k6-utils/1.4.0/index.js'

export const randomUser = () => ({
    id: uuidv4(),
    name: randomString(35),
})

export const randomBook = () => ({
    id: uuidv4(),
    title: randomString(35),
    author: randomString(35)
})

export const randomOffer = (bookId, sellerId) => ({
    id: uuidv4(),
    bookId: bookId,
    sellerId: sellerId,
    price: randomIntBetween(1, 99),
    stock: randomIntBetween(1, 5)
})
